containers/Dockerfile.build

```diff
@@ -10,21 +10,21 @@ ARG COMPILER_NAME
 ARG COMPILER_PACKAGE
 ARG COMPILER_VERSION
 
+LABEL au.org.access-nri.ci.compiler ${COMPILER_PACKAGE}@${COMPILER_VERSION}
+LABEL au.org.access-nri.ci.package ${PACKAGE}
+LABEL au.org.access-nri.ci.base-spack-iamge ${BASE_IMAGE}
+
 # Use Spack shell environment for subsequent RUN steps
 SHELL ["docker-shell"]
 
 # Install required compiler
-RUN spack -d install ${COMPILER_PACKAGE}@${COMPILER_VERSION}
-
 # Load required compiler and generate site-wide compilers.yaml
-RUN spack load ${COMPILER_PACKAGE}@${COMPILER_VERSION} && \
-  spack compiler find --scope site
-
 # Build and install package dependencies
-RUN spack -d install --only dependencies ${PACKAGE}%${COMPILER_NAME}@${COMPILER_VERSION}
+RUN spack -d install ${COMPILER_PACKAGE}@${COMPILER_VERSION} \
+ && spack load ${COMPILER_PACKAGE}@${COMPILER_VERSION} \
+ && spack compiler find --scope site \
+ && spack -d install --only dependencies ${PACKAGE}%${COMPILER_NAME}@${COMPILER_VERSION} \
+ && spack gc -y
```

Comment:
Can you comment (in the PR) why it is a good idea to do all this in a single `RUN` command. Just for general information for the morbidly curious.

Reply:
So, as far as I understand it:
* Every time you execute a `RUN` command, it creates a new 'layer' on top of the image. New layers have a small space overhead when they are created. 
* When stuff is added _or modified or deleted_ in a layer, this takes up space, as later layers are essentially a delta of what was, and what was executed in the `RUN` command. 
  * So, for example, if we do `RUN dnf install -y complex_package`, it will install the package and all the dependencies, and some of the cruft that comes with installing things.  
  * If we then do `RUN dnf clean all`, it will remove some of the cruft and remove unneeded packages. But, crucially, the final image could be _the same size or larger_ after this, as the delta of installing cruft and then removing it across `RUN` commands makes it larger.
* So, if we do all the `spack` commands in one layer, including `spack gc` and other cleanup (garbage collection), it will be smaller as all the commands are being executed in the same layer, so the delta will be free of any add-then-later-remove parts of the filesystem.